### PR TITLE
Fix icon picker reset by hass update

### LINKED ISSUE #13550: Entity Icon Filter/Search not working
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When attempting to change and Entities' Icon in it's setting, attempting to filter/search the icons is not usable, it shows the results for a flash then resets to the top of the list. Tested iOS app and Chrome.

https://user-images.githubusercontent.com/11084412/187907747-22fcc57c-6a29-409e-83fc-fb38329c9d95.mov

### Describe the behavior you expected

list shows filtered items

### Steps to reproduce the issue

1. go to an entities settings
2. click in Icon box 
3. type something
...

### What version of Home Assistant Core has the issue?

2022.9.0b1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

iOS 15.6.1 (HA App/Safari) / Chrome 104.0.5112.101

### Which operating system are you using to run this browser?

iOS and Mac 12.5.1

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Do not update if only `hass` changes.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #13550 
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-icon-picker.ts [hunk 1]
```diff
@@ -123,6 +123,14 @@ export class HaIconPicker extends LitElement {
     }
   }
 
+  protected shouldUpdate(changedProps: PropertyValues) {
+    const keys = Array.from(changedProps.keys());
+    if (keys.length === 1 && keys[0] === "hass" && this.hass) {
+      return false;
+    }
+    return true;
```
Comment (bramkragten) [inline src/components/ha-icon-picker.ts:131]:
```suggestion
    return !(changedProps.size === 1 && changedProps.has("hass"));
```

**Comment by bramkragten** [comment]
I think we should also add `_opened` to the `shouldUpdate` so it just doesnt update when open.

**Comment by piitaya** [comment]
I think only `_opened` is enough.
```ts
protected shouldUpdate() {
  return !this._opened;
}
```
What do you think @bramkragten?

**Comment by bramkragten** [comment]
Yes, that would be enough

**Review by bramkragten — APPROVED** [review]

**Pull request merged**